unit/java_bytecode/generate_concrete_generic_type/generate_java_generic_type.cpp

```diff
@@ -0,0 +1,252 @@
+/*******************************************************************\
+
+ Module: Unit tests for generating new type with generic parameters
+         substitued for concrete types
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <catch.hpp>
+
+#include <util/symbol_table.h>
+
+#include <java_bytecode/generate_java_generic_type.h>
+#include <java_bytecode/java_types.h>
+
+#include <iostream>
+#include <java_bytecode/java_bytecode_language.h>
+#include <language.h>
+
+SCENARIO(
+  "generate_java_generic_type_custom_type",
+  "[CORE][java_bytecode][generate_java_generic_type]")
+{
+  std::unique_ptr<languaget> java_lang(new_java_bytecode_language());
+
+  // Configure the language
+  null_message_handlert message_handler;
+  java_lang->set_message_handler(message_handler);
```

Comment:
I don't think `java_lang` and `java_code_stream` are used anywhere in this test.